hydra/core/utils.py

```diff
@@ -166,14 +173,29 @@ def register(name: str, f: Any) -> None:
     register("python_version", lambda level="minor": version_dict.get(level))
 
 
+class JobResult(Enum):
+    COMPLETED = 1
+    FAILED = 2
+
+
 @dataclass
 class JobReturn:
     overrides: Optional[Sequence[str]] = None
-    return_value: Any = None
     cfg: Optional[DictConfig] = None
     hydra_cfg: Optional[DictConfig] = None
     working_dir: Optional[str] = None
     task_name: Optional[str] = None
+    job_result: Optional[JobResult] = None
+    _return_value: Any = None
+
+    @property
+    def return_value(self) -> Any:
+        if self.job_result is None:
+            raise ValueError("job_result not yet available.")
```

Comment:
This should never happen. an assertion makes more sense.
```suggestion
        assert self.job_result is not None, "job_result not yet available"
```

If not an assertion, then I think we should use some other exception type. ValueError does not seem right here.